packages/dapp/src/pages/EscrowInfo.jsx

```diff
@@ -81,19 +83,25 @@ export const EscrowInfo = () => {
   };
 
   const getSmartInvoiceData = async () => {
-    if (
-      utils.isAddress(context.invoice_id) &&
-      !Number.isNaN(parseInt(context.chainID))
-    ) {
-      let smartInvoice = await getSmartInvoiceAddress(
-        context.invoice_id,
-        context.provider
-      );
-      console.log(context.chainID, smartInvoice);
-      getInvoice(parseInt(context.chainID), smartInvoice).then((i) => {
-        console.log(i);
-        setInvoice(i);
-      });
+    try {
+      if (
+        utils.isAddress(context.invoice_id) &&
+        !Number.isNaN(parseInt(context.chainID))
+      ) {
+        let smartInvoice = await getSmartInvoiceAddress(
+          context.invoice_id,
+          context.provider,
+        );
+        console.log(context.chainID, smartInvoice);
+        getInvoice(parseInt(context.chainID), smartInvoice).then(i => {
```

Comment:
Could we instead do `const invoice = await getInvoice(...);`
then handle it ?

Reply:
imo we should ether do `promise.then` everywhere or use `awawit` and `async`.
Here we are doing both, and seems odd to me.

Reply:
Sounds right. Will change it.

Reply:
updated it.